components/ai_chat/core/browser/engine/oai_api_client.cc

```diff
@@ -151,13 +170,26 @@ void OAIAPIClient::OnQueryCompleted(GenerationCompletedCallback callback,
     return;
   }
 
+  if (last_error_.has_value()) {
+    std::move(callback).Run(base::unexpected(last_error_.value()));
+    last_error_.reset();
+    return;
+  }
+
   // Handle error
-  std::move(callback).Run(base::unexpected(mojom::APIError::ConnectionIssue));
+  std::move(callback).Run(base::unexpected(
+      mojom::APIError(mojom::APIErrorType::ConnectionIssue, std::nullopt)));
```

Comment:
This case could be merged with the above if we stored a `std::optional<std::string> last_error_message_` on the class:

```
std::move(callback).Run(base::unexpected(mojom::APIError(mojom::APIErrorType::ConnectionIssue, last_error_)));
last_error_.reset();
```

Reply:
your suggestion works for now because both areas have the same error types, but in the future, if we require more granular error types based on the API's error messaging, this will need to change. that being said, good catch. i have revised this line.